src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs

```diff
@@ -427,7 +427,10 @@ public static void Interrupt(Thread thread)
 
         public static void OnThreadExiting(Thread thread)
         {
-            thread.WaitInfo.OnThreadExiting();
+            // There is a race around WaitInfo initialization, and so this check is a temporary measure to avoid NREs on thread destruction
+            // https://github.com/dotnet/runtime/issues/49627
+            if (thread.WaitInfo != null)
```

Comment:
I think it would be better for this null check to be in `~Thread`. 

We can even get rid of this method and just call the instance `OnThreadExiting` method directly:

```
        ~Thread()
        {
#if TARGET_UNIX || TARGET_BROWSER
            _waitInfo?.OnThreadExiting();
#endif
...
```

Reply:
I assume we want to leave the intermediary method for CoreRT (assuming you're continuing to call it) but just not use it here? Or should I get rid of it altogether?

Reply:
Get rid of it altogether. We will update CoreRT as needed.